app/helpers/application_helper.rb

```diff
@@ -1,7 +1,16 @@
 module ApplicationHelper
   def signup_link
-    unless current_page?(signup_path)
-      content_tag(:li, (link_to "Sign Up", signup_path))
-    end
+    return if current_user || current_page?(signup_path)
+    content_tag(:li, (link_to "Sign Up", signup_path))
+  end
+  
+  def login_link
+    return if current_user || current_page?(login_path)
+    content_tag(:li, (link_to "Log In", login_path))
+  end
+  
+  def logout_link
+    return unless current_user
+    content_tag(:li, (link_to "Log Out", logout_path, method: 'delete'))
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.